b2sdk/raw_simulator.py

```diff
@@ -402,7 +403,7 @@ def get_upload_url(self):
         return dict(bucketId=self.bucket_id, uploadUrl=upload_url, authorizationToken=upload_url)
 
     def get_upload_part_url(self, file_id):
-        upload_url = 'https://upload.example.com/part/%s' % (file_id,)
+        upload_url = 'https://upload.example.com/part/%s/%d' % (file_id, random.randint(1, 10**9))
```

Comment:
this will randomly cause a failure every 10 million executions or so. I'll fix it.

Reply:
actually more, because some tests may upload multiple files concurrently increasing the collision frequency